pkg/project/auth/cache.go

```diff
@@ -113,37 +114,42 @@ type AuthorizationCache struct {
 
 	reviewer Reviewer
 
-	namespaceInterface            kclient.NamespaceInterface
-	policyBindingsNamespacer      client.PolicyBindingsNamespacer
-	policiesNamespacer            client.PoliciesNamespacer
-	clusterPolicyBindingInterface client.ClusterPolicyBindingsInterface
-	clusterPolicyInterface        client.ClusterPoliciesInterface
+	namespaceInterface       kclient.NamespaceInterface
+	policyBindingsNamespacer client.PolicyBindingsNamespacer
+	policiesNamespacer       client.PoliciesNamespacer
+
+	policyCache authpolicy.PolicyCacheAdapter
 
 	syncHandler func(request *reviewRequest, userSubjectRecordStore cache.Store, groupSubjectRecordStore cache.Store, reviewRecordStore cache.Store) error
 }
 
 // NewAuthorizationCache creates a new AuthorizationCache
-func NewAuthorizationCache(reviewer Reviewer,
-	namespaceInterface kclient.NamespaceInterface, policyBindingsNamespacer client.PolicyBindingsNamespacer, policiesNamespacer client.PoliciesNamespacer,
-	clusterPolicyBindingInterface client.ClusterPolicyBindingsInterface, clusterPolicyInterface client.ClusterPoliciesInterface) *AuthorizationCache {
+func NewAuthorizationCache(reviewer Reviewer, namespaceInterface kclient.NamespaceInterface, policyBindingsNamespacer client.PolicyBindingsNamespacer,
+	policiesNamespacer client.PoliciesNamespacer, policyCache authpolicy.PolicyCacheAdapter) *AuthorizationCache {
 	result := &AuthorizationCache{
-		namespaceStore:                cache.NewStore(cache.MetaNamespaceKeyFunc),
-		policyIndexer:                 cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{"namespace": cache.MetaNamespaceIndexFunc}),
-		policyBindingIndexer:          cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{"namespace": cache.MetaNamespaceIndexFunc}),
-		clusterPolicyStore:            cache.NewStore(cache.MetaNamespaceKeyFunc),
+		namespaceStore:     cache.NewStore(cache.MetaNamespaceKeyFunc),
+		namespaceInterface: namespaceInterface,
+
+		policyIndexer:      *policyCache.GetPolicyIndexer(),
+		policiesNamespacer: policiesNamespacer,
+
+		policyBindingIndexer:     *policyCache.GetPolicyBindingIndexer(),
+		policyBindingsNamespacer: policyBindingsNamespacer,
+
+		clusterPolicyStore:           cache.NewStore(cache.MetaNamespaceKeyFunc),
```

Comment:
this is something that we should pull from the PolicyCache Indexers... an Index is a Store, so look through to see how clusterPolicyStore was being used and see if we can swap an indexer.